QuizDifferently.js

```diff
@@ -1,67 +1,101 @@
-    // represented by 11, 12, 13, and 14 respectively.
-    // Ex. [
-    //  {suit: "hearts", rank: 4},
-    //  {suit: "spades", rank: 2},    
-    //  {suit: "diamonds", rank: 14},
-    //  {suit: "clubs", rank: 12}
-    // ]
-    // Return the array sorted by suit and rank
-    // Disclaimer: This is a tricky one to do with JavaScript
-    sortDeckOfCards(cards){
-        return cards;
-    }
+  // Question 3
+  // Given a shuffled deck of cards, sort them by rank and suit.
+  // To represent the deck of cards, you will recieve an array of objects. Each object will have a rank and a suit. J, Q, K, and A will be
+  // represented by 11, 12, 13, and 14 respectively.
+  // Ex. [
+  //  {suit: "hearts", rank: 4},
+  //  {suit: "spades", rank: 2},
+  //  {suit: "diamonds", rank: 14},
+  //  {suit: "clubs", rank: 12}
+  // ]
+  // Return the array sorted by suit and rank
+  // Disclaimer: This is a tricky one to do with JavaScript
+  sortDeckOfCards(cards) {
+    return cards;
+  }
 
-    // Question 4
-    // Given a pile of socks, find how many socks are missing a pair.
-    // To represent the pile of socks, an array containing all the socks will be provided. Each sock will be represented by its color.
-    // Ex. ["Brown", "Blue", "Black", "Black", "Green", "Blue"]
-    // Return the number of socks without a match
-    missingMatches(socks){
-        return -1;
-    }
+  // Question 4
+  // Given a pile of socks, find how many socks are missing a pair.
+  // To represent the pile of socks, an array containing all the socks will be provided. Each sock will be represented by its color.
+  // Ex. ["Brown", "Blue", "Black", "Black", "Green", "Blue"]
+  // Return the number of socks without a match
+  missingMatches(socks) {
+    return -1;
+  }
 
-    //Question 5
-    // Given a roster of names, remove the duplicates from the roster
-    // To represent the roster, an array containing all the names will be provided.
-    // Ex. ["Daniel", "Kaleb", "Zan", "Mark", "Tariq", "Mark"]
-    // Return the an arrary without any duplicates
-    removeDuplicates(names){
-        return names;
+  //Question 5
+  // Given a roster of names, remove the duplicates from the roster
+  // To represent the roster, an array containing all the names will be provided.
+  // Ex. ["Daniel", "Kaleb", "Zan", "Mark", "Tariq", "Mark"]
+  // Return the an arrary without any duplicates
+  removeDuplicates(names) {
+    let rosterSet = new Set();
+    let noDupsRoster = [];
+
+    //insert roster of names to a Set object
+    //no duplicates will be inserted into set object
+    for (let i = 0; i < names.length; i++) {
+      rosterSet.add(names[i]);
     }
+    //copy each name from the set into a new no duplicate name roster, "noDupsRoster"
+    rosterSet.forEach((name) => {
+      noDupsRoster.push(name);
+    });
+
+    //return names from no duplicate name roster, "noDupsRoster"
+    return noDupsRoster;
+  }
```

Comment:
Nice solution! I challenge you to use the `foreach` iteration method here as well